src/plugin.js

```diff
@@ -185,13 +185,23 @@ export const setupSessions = (player) => {
  */
 export const emeErrorHandler = (player) => {
   return (objOrErr) => {
-    const message = typeof objOrErr === 'string' ? objOrErr : (objOrErr && objOrErr.message) || null;
-
-    player.error({
+    const error = {
       // MEDIA_ERR_ENCRYPTED is code 5
-      code: 5,
-      message
-    });
+      code: 5
+    };
+
+    if (typeof objOrErr === 'string') {
+      error.message = objOrErr;
+    } else if (objOrErr) {
+      if (objOrErr.message) {
+        error.message = objOrErr.message;
+      }
+      if (objOrErr.cause && objOrErr.cause.length) {
+        error.cause = objOrErr.cause;
+      }
+    }
```

Comment:
I think it makes it a bit clearer what's happening, plus allows us to set both the cause and the message based on the potential inputs to this function.